### DISCUSSION TITLE
Last+first name, first+last name or just name?

### DISCUSSION THREAD

**Discussion by prschopf** [Larian77/Openbiblio#74] [Ideas]
We dont use last name+first name for members, here in Brazil. we use just "name" and insert first+last name there.

To solve that in Openbilio, Ive changed every code line from concat(last name, ',', first name) to concat(first name, ' ', last name).

so, I have a few questions:

- How is that kind of thing in your country?
- Is "last+first" used by most of countries in Europe?
- Would you change it to first+last if you had this option in Openbiblio?

**Comment by superudo** [comment]
Naming conventions vary by country and culture, but the cataloging convention of last name, first name exists for a practical reason that transcends any single culture: it makes sorted lists meaningful.                                                                                    

In a sorted list ordered by first name, you group together everyone named "Maria" or "John" — people who share nothing except a given name. Ordered by last name, you group families together, which is far more useful when browsing an author or patron index.

This is why library cataloging standards worldwide (ISBD, MARC, RDA) use last-name-first, regardless of how names are spoken naturally in a given country. Even in Brazil, a catalog entry for "José Silva" would conventionally be indexed as "Silva, José."

A configurable option for display order (showing "José Silva" in the UI while storing/sorting as "Silva, José") might be the best of both worlds — natural reading in one context, correct sorting in another. However, deciding on the right approach and implementing it properly takes considerable effort, so it is not a trivial change.
